oeps/architectural-decisions/oep-0065/decisions/0001-unified-platform-repository.rst

```diff
@@ -0,0 +1,109 @@
+
+The bulk of the work to build OEP-65's reference implementation is related to the *library repositories* that MFEs depend on:
+
+* frontend-build
+* frontend-platform
+* frontend-plugin-framework
+* frontend-component-header
+* frontend-component-footer
+
+It will also involve the creation of a new *shell application* to initialize the frontend at runtime and load existing MFEs as modules.
+
+There are several related concerns which influence our strategy for migrating these library repositories to the new module MFE architecture.
+
+Significant API changes
+=======================
+
+The creation of the shell application and the move toward using plugins and module federation is expected to significantly change our libraries' API surface as the shell takes responsibility for much of what ``frontend-platform`` provides to MFEs in the application MFE architecture.
+
+The development environment will also change, as we want to be able to run a single development server that loads the MFE under development into the shell, which means alternatives to frontend-build's existing helpers will need to be created.
+
+The header and footer will be owned by the shell and need not be libraries any more (caveat: maintaining good ways to customize them), and we expect ``frontend-plugin-framework`` to be folded into ``frontend-platform`` to unify the two, as they will be intimately tied to each other going forward.
+
+Reducing Dependency Maintenance
+===============================
+
+One of the motivators of OEP-65 was reducing dependency maintenance.  Not explicitly called out in the OEP is the desire to have fewer repositories to manage, which will in turn further reduce the number of dependencies in MFEs.  We see value in combining the library repositories listed above into a single library, which will in turn reduce the number of dependencies to manage in downstream repositories.
+
+Modernization
+=============
+
+We also feel the need to continue to modernize our library repositories by adopting industry standard technologies like Typescript, or more performant webpack loaders, and there's some sentiment that this may be the right time to make these changes as we're already undergoing a paradigm shift.  ``frontend-platform``, for instance, simplifies significantly if we use TypeScript types instead of the bespoke "interface" and "service implementation" system in that repository.
```

Comment:
> Typescript, ... more performant webpack loaders, ... simplifies significantly if we use TypeScript types instead of the bespoke "interface"

My hero ❤️